### ISSUE TITLE
uploads blocked by overly eager package checks

### ISSUE THREAD

**Issue by bos** [haskell/hackage-server#360]
The upstream bug is https://github.com/haskell/cabal/issues/2527.

The problem is that hackage has started blocking uploads of packages that conditionally use `-Werror` hidden behind a default-`False` flag, a pattern used widely in the community. I can't upload any of my numerous packages, for example.

**Comment by dcoutts** [comment]
Sorry about this, we hadn't intended to change it, or really realise that it had changed. It cam with the Cabal-1.22 version (though as I say, not clear that it was intentional).

We'll sort something out.

**Comment by bos** [comment]
I've resorted to hand-editing my `.cabal` files after tagging each release, so I'm not completely blocked.

**Comment by cartazio** [comment]
@dcoutts @hvr whats the status on fixing that uploading kerfuffle?

**Comment by hvr** [comment]
@cartazio afaik it's still blocked on haskell/cabal#2527

**Comment by tfausak** [comment]
I hit this problem even though I only use `-Werror` in my test suite: https://github.com/tfausak/flow/blob/v1.0.1/flow.cabal#L45.

**Comment by cartazio** [comment]
I thought we had the fix in cabal master.  Do we need a new hackage server
deploy?

**Comment by hvr** [comment]
@cartazio the fix is actually in and deployed, and if you look at haskell/cabal#2594 you'll see that it requires  `-Werror`-flags to be guarded behind a flag for *all* stanzas, including testsuites/benchmarks

**Comment by dcoutts** [comment]
Right, we deployed this already.

**Comment by bergmark** [comment]
Did this get reverted?

```
V~/r/aeson master> cabal upload dist/aeson-0.11.2.0.tar.gz
Hackage password:
Uploading dist/aeson-0.11.2.0.tar.gz...
Error uploading dist/aeson-0.11.2.0.tar.gz: http code 400
Error: Invalid package

'ghc-options: -Wall -Werror' makes the package very easy to break with future
GHC versions because new GHC versions often add new warnings. Use just
'ghc-options: -Wall' instead.
```

**Comment by bergmark** [comment]
I'll put this here as well for the record.

This was changed in https://github.com/haskell/cabal/commit/cbfea2ada2f2755c81b12b355456a9f411191f22 but was never merged into the `1.22` branch so it looks like it was just cherry-picked for hackage.

Since it seems cabal 1.24 will be released soon (https://github.com/haskell/cabal/issues/2527#issuecomment-215183486) I'd suggest we wait for that and then upgrade Cabal in hackage-server, does that sound good? For aeson I can disable -Werror completely for this release.

**Comment by dcoutts** [comment]
Thanks for investigating. Yes, we should be able to deploy with 1.24 very soon.

**Comment by hvr** [comment]
see also #497
